seaworthy/tests-core/test_logs.py

```diff
@@ -107,3 +111,149 @@ def test_str(self):
             str(matcher),
             "SequentialLinesMatcher(matched=[EqualsMatcher('foo'), "
             "RegexMatcher('^bar')], unmatched=[])")
+
+
+class FakeLogsContainer:
+    """
+    A container object stub that emits canned logs.
+
+    Logs can either be streamed or tailed. After a log entry has been streamed,
+    it is stored and we don't wait for it next time. Only logs that have
+    already been streamed are tailed.
+    """
+
+    def __init__(self, log_entries, expected_kw=None):
+        self.log_entries = log_entries
+        self._seen_logs = []
+        self._expected_kw = {} if expected_kw is None else expected_kw
+
+    def logs(self, stream=False, **kw):
+        if stream:
+            # We're streaming logs, so return our iterator.
+            assert kw == self._expected_kw
```

Comment:
This stub is only designed to be used through `_last_few_log_lines()` and `wait_for_logs_matching()`, so we don't really care about what happens if you pass extra kwargs when tailing. There aren't any tests for encodings or kwargs when streaming yet, either. I guess I should either write those or put in some placeholders.